source/slang/slang-parser.cpp

```diff
@@ -6411,8 +6411,26 @@ namespace Slang
         // At this point we can also parse bitwise or expressions
         //
         while(!(parser->LookAheadToken(TokenType::RBrace)
-            || parser->LookAheadToken(TokenType::Semicolon)))
+            || parser->LookAheadToken(TokenType::Semicolon))
```

Comment:
Whenever writing a loop like this, we need to consider the end of file case to avoid looping infinitely.

Reply:
Also consider following the pattern in `ParseBlockStatement` to setup skipping/error recovering.

Reply:
Ah, I thought this loop was ok as if we don't read a token we return from the function anyway